File: 230113-dojo/javascript/solution.js
export default function solution(d, budget) {
  const sumArray = d.sort((a, b) => a - b).map((_, index) => d.slice(0, index + 1)
    .reduce((pre, cur) => pre + cur, 0));

  return sumArray.filter((value) => value <= budget).length;
}

File: 230113-dojo/java/app/src/main/java/Solution.java
import java.util.*;

class Solution {
    public int solution(int[] d, int budget) {
        int sumOfArray = Arrays.stream(d).sum();

        if (sumOfArray <= budget) {
            return d.length;
        }

        return process(d, budget);
    }

    public int process(int[] d, int budget) {
        Arrays.sort(d);

        for (int i = 0; i < d.length; i += 1) {
            budget -= d[i];

            if (budget > 0) {
                continue;
            }

            if (budget == 0) {
                return i + 1;
            }

            return i;
        }

        return 0;
    }
}
